### PR TITLE
Changes in gtsam to support hybrid branches

### PR DESCRIPTION
**Pull request by dellaert**
This PR collect a number of commits to files made in the hybrid development branches.
I made the commits below descriptive.
@varunagrawal we can re-target the PRs or just wait until this is merged.

### CODE REVIEW

File: gtsam/discrete/DecisionTree.h
Comment (dellaert) [inline gtsam/discrete/DecisionTree.h:345]: Actually, @ProfFan , this should probably have been a static method and be called `Unzip`. Or a method, and then it would not take an argument, right?

Comment (dellaert) [reply gtsam/discrete/DecisionTree.h:345]: BTW, just looking for consensus, I think I would change it to static.

**Comment by dellaert** [comment]
Killed CI while deciding about Unzip

File: gtsam/discrete/DecisionTree.h
Comment (dellaert) [reply gtsam/discrete/DecisionTree.h:345]: Actually, I see now it is neither: it is a free function :-/ Which is correctly named, so never mind! Will re-format is all.

Comment (ProfFan) [reply gtsam/discrete/DecisionTree.h:345]: Do you want to reopen CI?

Comment (dellaert) [reply gtsam/discrete/DecisionTree.h:345]: Yes, I'll just do the required one and merge

**Comment by dellaert** [comment]
I think this is ready to merge if I can get an approving review from @ProfFan or @varunagrawal

**Comment by varunagrawal** [comment]
Sorry I was out the entire afternoon. I'll review in the next 30 minutes.

**Review by varunagrawal — APPROVED** [review]
LGTM! Merge whenever.

**Comment by dellaert** [comment]
I can't merge because there is some CI waiting to be reported. Any ideas?

**Comment by ProfFan** [comment]
I reopened the PR, now it should start to build. @dellaert

**Comment by ProfFan** [comment]
@dellaert CI is ready :)

**Pull request merged**
